From 27e22791322cd1a4cf9004405b56fb6dd1ab824d Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Fri, 7 Jun 2013 20:39:08 +0100
Subject: [PATCH 169/227] FE: Omit call to WriteDefaults during startup.

WriteDefaults is supposed to force the creation of missing settings
in the database.  Unfortunately the time overhead is quite large on
remote FE's and it's largely pointless.  Missing settings will assume
default values at runtime and in any case will be written to the dbase
when the user next invokes the relevant settings page.

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/programs/mythfrontend/main.cpp |   11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/mythtv/programs/mythfrontend/main.cpp b/mythtv/programs/mythfrontend/main.cpp
index e444ca6..4106566 100644
--- a/mythtv/programs/mythfrontend/main.cpp
+++ b/mythtv/programs/mythfrontend/main.cpp
@@ -1099,6 +1099,14 @@ static bool RunMenu(QString themedir, QString themename)
 
 // If any settings are missing from the database, this will write
 // the default values
+//
+// This is pointless and very time consuming, especiall on a remote FE.
+// Any missing settings will assume default values at runtime and
+// in any case will be written to the dbase when the user next invokes
+// the relevant settings page.
+//
+// This would be more useful for _resetting_ to known default values
+#if 0
 static void WriteDefaults()
 {
     PlaybackSettings ps;
@@ -1129,6 +1137,7 @@ static void WriteDefaults()
     rsettings.Load();
     rsettings.Save();
 }
+#endif
 
 static int internal_play_media(const QString &mrl, const QString &plot,
                         const QString &title, const QString &subtitle,
@@ -1700,7 +1709,7 @@ int main(int argc, char **argv)
         return GENERIC_EXIT_DB_OUTOFDATE;
     }
 
-    WriteDefaults();
+    //WriteDefaults();
 
     // Refresh Global/Main Menu keys after DB update in case there was no DB
     // when they were written originally
-- 
1.7.9.5

